src/client/client.go

```diff
@@ -558,13 +583,20 @@ func newOnUserMachine(cfg *config.Config, context *config.Context, contextName,
 
 	var fw *PortForwarder
 	if pachdAddress == nil && context.PortForwarders != nil {
-		pachdLocalPort, ok := context.PortForwarders["pachd"]
-		if ok {
-			log.Debugf("Connecting to explicitly port forwarded pachd instance on port %d", pachdLocalPort)
-			pachdAddress = &grpcutil.PachdAddress{
-				Secured: false,
-				Host:    "localhost",
-				Port:    uint16(pachdLocalPort),
+		if !CheckPortForwardLiveness(context) {
+			context.PortForwarders = nil
+			if err = cfg.Write(); err != nil {
+				return nil, errors.Wrap(err, "could not write config to remove port-forward ports")
```

Comment:
part of me is a little nervous about blocking every command if you somehow make your config non-writable after the port-forwarder is killed, but it doesn't seem very likely

Reply:
This reservation makes sense. One alternative that occurred to me is that this code could just warn users that they need to restart `pachctl port-forward`, though an issue is that if `pachctl port-forward` also can't modify the config, then other `pachctl` commands will have no way to use it—they'll keep checking the old pid for liveness, keep re-discoving that the old port-forward process is dead, and keep doing internal port-forwarding even though a new port-forward process has come up as a replacement.

I'm not really sure what a good solution would be. I was planning to merge this PR if/when the tests are green, but I'd be happy to revise this here (if you know what to do) or in a followup PR if I merge before you respond—just let me know.